File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    l = [0 for _ in range(n+1)]
    l[0] = 1
    count = 0
    for i in section:
        if l[i] == 0:
            if len(l) >= i+m:
                for j in range(i,i+m):
                    l[j] = 1
                count += 1
            else:
                for j in range(i,len(l)):
                    l[j] = 1
                count += 1
    return count

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        
        int answer = 1;
        int startIndex = section[0];
        int endIndex = startIndex + m - 1;
        
        for (int i = 0; i < section.length; i++) {
            if (section[i] > endIndex) {
                answer++;
                startIndex = section[i];
                endIndex = startIndex + m - 1;
            }
        }
            
        return answer;
    }
}